contracts/extensions/DeltaNeutralBasisTradingStrategyExtension.sol

```diff
@@ -0,0 +1,1456 @@
+     */
+    function _isAdvantageousTWAP(int256 _currentLeverageRatio) internal view returns (bool) {
+        uint256 twapLeverageRatioAbs = twapLeverageRatio.abs();
+        uint256 targetLeverageRatioAbs = methodology.targetLeverageRatio.abs();
+        uint256 currentLeverageRatioAbs = _currentLeverageRatio.abs();
+
+        return (
+            (twapLeverageRatioAbs < targetLeverageRatioAbs && currentLeverageRatioAbs >= twapLeverageRatioAbs)
+            || (twapLeverageRatioAbs > targetLeverageRatioAbs && currentLeverageRatioAbs <= twapLeverageRatioAbs)
+        );
+    }
+
+    /**
+     * Transfer ETH reward to caller of the ripcord function. If the ETH balance on this contract is less than required
+     * incentive quantity, then transfer contract balance instead to prevent reverts.
+     *
+     * return uint256           Amount of ETH transferred to caller
+     */
+    function _transferEtherRewardToCaller(uint256 _etherReward) internal returns(uint256) {
+        uint256 etherToTransfer = _etherReward < address(this).balance ? _etherReward : address(this).balance;
+
+        msg.sender.transfer(etherToTransfer);
+
+        return etherToTransfer;
+    }
+
+    /**
+     * Internal function returning the ShouldRebalance enum used in shouldRebalance and shouldRebalanceWithBounds external getter functions
+     *
+     * return ShouldRebalance         Enum detailing whether to rebalance, iterateRebalance, ripcord or no action
+     */
+    function _shouldRebalance(
+        int256 _currentLeverageRatio,
+        int256 _minLeverageRatio,
+        int256 _maxLeverageRatio
+    )
+        internal
+        view
+        returns(ShouldRebalance)
+    {
+        // Get absolute value of current leverage ratio
+        uint256 currentLeverageRatioAbs = _currentLeverageRatio.abs();
+
+        // If above ripcord threshold, then check if incentivized cooldown period has elapsed
+        if (currentLeverageRatioAbs >= incentive.incentivizedLeverageRatio.abs()) {
+            if (lastTradeTimestamp.add(incentive.incentivizedTwapCooldownPeriod) < block.timestamp) {
+                return ShouldRebalance.RIPCORD;
+            }
+            return ShouldRebalance.NONE;
+        }
+
+        // If TWAP, then check if the cooldown period has elapsed
+        if (twapLeverageRatio != 0) {
+            if (lastTradeTimestamp.add(execution.twapCooldownPeriod) < block.timestamp) {
+                return ShouldRebalance.ITERATE_REBALANCE;
+            }
+            return ShouldRebalance.NONE;
+        }
+
+        // If not TWAP, then check if the rebalance interval has elapsed OR current leverage is above max leverage OR current leverage is below
+        // min leverage
+        if (
+            block.timestamp.sub(lastTradeTimestamp) > methodology.rebalanceInterval
+            || currentLeverageRatioAbs > _maxLeverageRatio.abs()
+            || currentLeverageRatioAbs < _minLeverageRatio.abs()
```

Comment:
Maybe these comparisons could be moved to an `isWithinLeverageBounds` helper and used here, L1407, and L1436.

Reply:
L1436 uses and operation, while L1407 uses or operation. Skipped.

Reply:
L1430 is `!withinLeverageBounds` and L1459 is `withinLeverageBounds`?

https://github.com/SetProtocol/set-v2-strategies/blob/779642f691107a35b67c094b3d03d471c17992bf/contracts/extensions/DeltaNeutralBasisTradingStrategyExtension.sol#L1430-L1431

https://github.com/SetProtocol/set-v2-strategies/blob/779642f691107a35b67c094b3d03d471c17992bf/contracts/extensions/DeltaNeutralBasisTradingStrategyExtension.sol#L1459-L1460